lib/search_resolver.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require 'ostruct'
+
+# Returns a search result to the GraphQL API
+class SearchResolver
+  def call(root, arguments, _context)
+    query = root[:query]
+    categories = arguments[:categories]
+    if categories.blank?
+      indices = [::Index::RepositoryIndex::Repository,
+                  ::Index::OrganizationIndex::Organization,
+                  ::Index::UserIndex::User]
+    else
+      indices = categories.map do |category|
```

Comment:
Check out [`Enumerable#reduce`](https://ruby-doc.org/core-2.1.0/Enumerable.html#method-i-reduce):

```ruby
indices = categories.reduce do |indices, category|
  ...
    when 'repositories'
      indices + [::Index::RepositoryIndex::Repository
  ...
end
```

that way you don't need to flatten the array at the end, which might make it faster.